File: WeatherGoat/Models/Point.cs
using System.Text.Json.Serialization;

namespace WeatherGoat.Models;

public record Point
{
    [JsonPropertyName("gridX")]
    public int GridX { get; set; }

    [JsonPropertyName("gridY")]
    public int GridY { get; set; }
    
    [JsonPropertyName("forecast")]
    public string Forecast { get; set; }
    
    [JsonPropertyName("forecastHourly")]
    public string HourlyForecast { get; set; }
    
    [JsonPropertyName("forecastGridData")]
    public string ForecastGridData { get; set; }
    
    [JsonPropertyName("observationStations")]
    public string ObservationStations { get; set; }
    
    [JsonPropertyName("relativeLocation")]
    public RelativeLocation RelativeLocation { get; set; }
    
    [JsonPropertyName("forecastZone")]
    public string ForecastZone { get; set; }
    
    [JsonPropertyName("county")]
    public string? County { get; set; }
    
    [JsonPropertyName("fireWeatherZone")]
    public string FireWeatherZone { get; set; }
    
    [JsonPropertyName("timeZone")]
    public string Timezone { get; set; }
    
    [JsonPropertyName("radarStation")]
    public string RadarStation { get; set; }

    public string RadarImageUrl => $"https://radar.weather.gov/ridge/standard/{RadarStation}_loop.gif";
    public string CountyId      => County != null ? County.Split('/')[^1] : string.Empty;
    public string ZoneId        => ForecastZone.Split('/')[^1];
}

File: src/models/point.ts
import { Type } from 'class-transformer';
import { RelativeLocation } from '@models/relative-location';

export class Point {
	public gridX!: number;
	public gridY!: number;
	public forecast!: string;
	public forecastHourly!: string;
	public forecastGridData!: string;
	public observationStations!: string;
	@Type(() => RelativeLocation)
	public relativeLocation!: RelativeLocation;
	public forecastZone!: string;
	public county?: string;
	public fireWeatherZone!: string;
	public timeZone!: string;
	public radarStation!: string;

	public get countyId() {
		return this.county ? this.extractLastUrlSegment(this.county) : '';
	}

	public get zoneId() {
		return this.extractLastUrlSegment(this.forecastZone);
	}

	public get radarImageUrl() {
		return `https://radar.weather.gov/ridge/standard/${this.radarStation}_loop.gif`;
	}

	private extractLastUrlSegment(url: string) {
		const segments = url.split('/');
		return segments[segments.length - 1]!;
	}
}
